### PR TITLE
Superadmin product restructure

### PR DESCRIPTION
**Pull request by SeefatHimel**
# Description
(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change
- [x] UI Fix
- [x]  Restructure

(Please delete options that are not relevant.)

- [x] Bug fix
- [x] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [ ] Separation of Concerns followed (Reviewer name/NaN)
- [ ] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/superadmin/components/products/editProductDetails.tsx [hunk 1]
```diff
@@ -121,15 +161,17 @@ const EditProduct = (props: any) => {
...
     const id = product.id;
     if (categories[0]) {
-      const response = await userAPI.updateProduct(newData, id , router);
+      const response = await userAPI.updateProduct(newData, id, router);
     } else toast.error("You must select a cateory");
   };
 
   const getCategoryData = () => {
-    categogiesData.map((category: any, index) => {
+    categogiesData.map((category: CategoryInterface, index) => {
```
Comment (utpaulBS23) [inline packages/superadmin/components/products/editProductDetails.tsx]: declar type

File: packages/superadmin/APIs/restApi.ts [hunk 2]
```diff
@@ -200,7 +203,7 @@ export async function getManufacturerRest(
   try {
     const { data } = await axios?.get(`${apiEndPoints?.manufacturerList}`);
     return data?.data as Manufacturer[];
-  } catch (error) {
+  } catch (error: any) {
```
Comment (utpaulBS23) [inline packages/superadmin/APIs/restApi.ts:206]: remove any and place exact error type

File: packages/superadmin/APIs/restApi.ts [hunk 3]
```diff
@@ -216,8 +219,8 @@ export async function createManufacturerRest(
     router.push('/Manufacturer/');
     toast.success('Create Successful');
     return response.data as CreateManufacturerRequest;
-  } catch (error) {
-    console.log(error);
+  } catch (error: any) {
```
Comment (utpaulBS23) [inline packages/superadmin/APIs/restApi.ts:222]: same in previous commit

File: packages/superadmin/APIs/restApi.ts [hunk 4]
```diff
@@ -233,7 +236,7 @@ export async function deleteManufacturerRest(
     router.push('/Manufacturer/');
     toast.success('Successfully deleted');
     return data?.data as Manufacturer[];
-  } catch (error) {
+  } catch (error: any) {
```
Comment (utpaulBS23) [inline packages/superadmin/APIs/restApi.ts:239]: same

File: packages/superadmin/components/products/createProduct.tsx [hunk 5]
```diff
@@ -1,17 +1,47 @@
...
+import CategoryForm from '@/components/products/forms/categoryForm';
+import ProductInfoForm from '@/components/products/forms/productInfoForm';
 
-import { useRouter } from "next/router";
-import CategoryForm from "./forms/categoryForm";
-import { useEffect, useState } from "react";
-import { toast } from "react-toastify";
-
-const CreateProduct = () => {
+interface FormDataInterFace {
```
Comment (utpaulBS23) [inline packages/superadmin/components/products/createProduct.tsx]: make models folder and place this interface on it

File: packages/superadmin/components/products/forms/categoryForm.tsx [hunk 6]
```diff
@@ -1,18 +1,33 @@
...
+  isSelected: boolean;
+  isFeatured: boolean;
+  displayOrder: number;
+}
+interface CategoryFormInterface {
+  setCategoryData: Function;
+  categoryData: CategoryInterface[];
+  setFieldValue: Function;
+}
+
```
Comment (utpaulBS23) [inline packages/superadmin/components/products/forms/categoryForm.tsx]: same as previous commit

File: packages/superadmin/components/products/forms/categoryForm.tsx [hunk 7]
```diff
@@ -24,22 +39,22 @@ const CategoryForm = (props: any) => {
...
     if (categoryID == 0) {
       toast.error("Please Select a Category");
       return;
     }
-    const isFeatured = document.getElementById("isFeaturedCategory").value;
-    const displayOrder = document.getElementById("displayOrderCategory").value;
-    const newcat = categoryData.map((data: any, index: any) => {
+    const isFeatured = document.getElementById("isFeaturedCategory")?.value;
+    const displayOrder = document.getElementById("displayOrderCategory")?.value;
+    categoryData.map((data: CategoryInterface, index: any) => {
```
Comment (utpaulBS23) [inline packages/superadmin/components/products/forms/categoryForm.tsx]: Why index is any

**Review by utpaulBS23 — CHANGES_REQUESTED** [review]
please make frontend model seperate

File: packages/superadmin/APIs/restApi.ts (hunk 4 above)
Comment (SeefatHimel) [reply packages/superadmin/APIs/restApi.ts:239]: cant put anything but any

File: packages/superadmin/APIs/restApi.ts (hunk 3 above)
Comment (SeefatHimel) [reply packages/superadmin/APIs/restApi.ts:222]: cant put anything but any

File: packages/superadmin/APIs/restApi.ts (hunk 2 above)
Comment (SeefatHimel) [reply packages/superadmin/APIs/restApi.ts:206]: cant put anything but any

File: packages/superadmin/components/products/createProduct.tsx (hunk 5 above)
Comment (SeefatHimel) [reply packages/superadmin/components/products/createProduct.tsx]: Done

File: packages/superadmin/components/products/editProductDetails.tsx (hunk 1 above)
Comment (SeefatHimel) [reply packages/superadmin/components/products/editProductDetails.tsx]: Done

File: packages/superadmin/components/products/forms/categoryForm.tsx (hunk 6 above)
Comment (SeefatHimel) [reply packages/superadmin/components/products/forms/categoryForm.tsx]: Done

File: packages/superadmin/components/products/forms/categoryForm.tsx (hunk 7 above)
Comment (SeefatHimel) [reply packages/superadmin/components/products/forms/categoryForm.tsx]: changed

**Pull request merged**